x-pack/plugins/siem/public/management/pages/policy/view/policy_list.test.tsx

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import * as reactTestingLibrary from '@testing-library/react';
+
+import { PolicyList } from './index';
+import { mockPolicyResultList } from '../store/policy_list/mock_policy_result_list';
+import { AppContextTestRender, createAppRootMockRenderer } from '../../../../common/mock/endpoint';
+import { AppAction } from '../../../../common/store/actions';
+
+describe('when on the policies page', () => {
+  let render: () => ReturnType<AppContextTestRender['render']>;
+  let history: AppContextTestRender['history'];
+  let store: AppContextTestRender['store'];
+
+  beforeEach(() => {
+    const mockedContext = createAppRootMockRenderer();
+    ({ history, store } = mockedContext);
+    render = () => mockedContext.render(<PolicyList />);
+  });
+
+  it('should show a table', async () => {
+    const renderResult = render();
+    const table = await renderResult.findByTestId('policyTable');
+    expect(table).not.toBeNull();
+  });
+
+  describe('when list data loads', () => {
+    let firstPolicyID: string;
+    beforeEach(() => {
+      reactTestingLibrary.act(() => {
+        history.push('/management/policy');
+        reactTestingLibrary.act(() => {
+          const policyListData = mockPolicyResultList({ total: 3 });
+          firstPolicyID = policyListData.items[0].id;
+          const action: AppAction = {
```

Comment:
I have had some issues with doing a dispatch like this before - because even though you are adding data to the store, the middleware will still issue an api call for the data because of the url changing action - so there is a chance that your state will be reset (ex. API call fails because no mock is setup for the API call, which cause a Server Filed To data action to be dispatched. This issue might be seen if a test case is done where it causes a "wait" to occur in the test code).

I think the right path is to setup a mock that handles API requests and add that to the `core.http.get|post` methods. Perhaps this is can be done later, but wanted to highlight it.